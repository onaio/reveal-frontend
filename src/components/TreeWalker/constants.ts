export const ACTIVE = 'Active';
export const FEATURE = 'Feature';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_CHILDREN = 'Could not load children';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
export const NOT_AVAILABLE = 'N/A';
