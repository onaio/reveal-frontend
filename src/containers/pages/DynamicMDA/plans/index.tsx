import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import {
  HIDDEN_PLAN_STATUSES,
  SUPERSET_DYNAMIC_MDA_REPORTING_PLANS_SLICE,
} from '../../../../configs/env';
import { MDA_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPLansReducer, {
  genericFetchPlans,
  makeGenericPlansArraySelector,
  reducerName as GenericPLansReducerName,
} from '../../../../store/ducks/generic/plans';
import { GenericPlan } from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(GenericPLansReducerName, GenericPLansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for Dynamic MDA plans */
const MDAPlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a list of MDA plans */
const MDAPLansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDAPLansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: MDA_PLANS,
  pageUrl: REPORT_MDA_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_DYNAMIC_MDA_REPORTING_PLANS_SLICE,
};

MDAPLansList.defaultProps = defaultProps;

export { MDAPLansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const plans = MDAPlansArraySelector(state, {
    interventionTypes: [InterventionType.DynamicMDA],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected MDA Plans component */
const ConnectedMDAPLansList = connect(mapStateToProps, mapDispatchToProps)(MDAPLansList);

export default ConnectedMDAPLansList;
