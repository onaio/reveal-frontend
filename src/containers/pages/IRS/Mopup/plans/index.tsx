import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { MOP_UP_REPORTING_TITLE } from '../../../../../configs/lang';
import { IRS_MOP_UP_REPORT_URL } from '../../../../../constants';
import { RouteParams } from '../../../../../helpers/utils';
import ConnectedIRSPlansList from '../../plans';

export const IRSMopUpReporting = (props: RouteComponentProps<RouteParams>) => {
  const plansProps = {
    ...props,
    pageTitle: MOP_UP_REPORTING_TITLE,
    pageUrl: IRS_MOP_UP_REPORT_URL,
  };
  return <ConnectedIRSPlansList {...plansProps} />;
};
