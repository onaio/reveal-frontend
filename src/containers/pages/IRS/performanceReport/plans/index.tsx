import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { IRS_PERFORMANCE_REPORTING_TITLE } from '../../../../../configs/lang';
import { PERFORMANCE_REPORT_IRS_PLAN_URL } from '../../../../../constants';
import { RouteParams } from '../../../../../helpers/utils';
import ConnectedIRSPlansList from '../../plans';

export const IRSPerfomenceReport = (props: RouteComponentProps<RouteParams>) => {
  const plansProps = {
    ...props,
    pageTitle: IRS_PERFORMANCE_REPORTING_TITLE,
    pageUrl: PERFORMANCE_REPORT_IRS_PLAN_URL,
  };
  return <ConnectedIRSPlansList {...plansProps} />;
};
