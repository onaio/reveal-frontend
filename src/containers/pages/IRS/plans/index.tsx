import ListView from '@onaio/list-view';
import reducerRegistry from '@onaio/redux-reducer-registry';
import React, { useEffect, useState } from 'react';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Col, Row } from 'reactstrap';
import { Store } from 'redux';
import HeaderBreadcrumb from '../../../../components/page/HeaderBreadcrumb/HeaderBreadcrumb';
import Loading from '../../../../components/page/Loading';
import { SUPERSET_IRS_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { HOME, HOME_URL, IRS_PLANS, REPORT_IRS_PLAN_URL } from '../../../../constants';
import supersetFetch from '../../../../services/superset';
import IRSPlansReducer, {
  fetchIRSPlans,
  getIRSPlansArray,
  IRSPlan,
  reducerName as IRSPlansReducerName,
} from '../../../../store/ducks/generic/plans';

/** register the plan definitions reducer */
reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);

/** interface for PlanList props */
interface PlanListProps {
  fetchPlans: typeof fetchIRSPlans;
  plans: IRSPlan[];
  service: typeof supersetFetch;
}

/** Simple component that loads the new plan form and allows you to create a new plan */
const IRSPlansList = (props: PlanListProps) => {
  const { fetchPlans, plans, service } = props;
  const [loading, setLoading] = useState<boolean>(true);

  const pageTitle: string = IRS_PLANS;

  const breadcrumbProps = {
    currentPage: {
      label: pageTitle,
      url: REPORT_IRS_PLAN_URL,
    },
    pages: [
      {
        label: HOME,
        url: HOME_URL,
      },
    ],
  };

  /** async function to load the data */
  async function loadData() {
    try {
      setLoading(plans.length < 1); // only set loading when there are no plans
      await service(SUPERSET_IRS_REPORTING_PLANS_SLICE).then((result: IRSPlan[]) =>
        fetchPlans(result)
      );
    } catch (e) {
      // do something with the error?
    } finally {
      setLoading(false);
    }
  }

  useEffect(() => {
    loadData();
  }, []);

  if (loading === true) {
    return <Loading />;
  }

  const listViewProps = {
    data: plans.map(planObj => {
      return [
        <Link to={`${REPORT_IRS_PLAN_URL}/${planObj.plan_id}`} key={planObj.plan_id}>
          {planObj.plan_title}
        </Link>,
        planObj.plan_date,
        planObj.plan_effective_period_start,
        planObj.plan_effective_period_end,
        planObj.plan_status,
      ];
    }),
    headerItems: ['Title', 'Date Created', 'Start Date', 'End Date', 'Status'],
    tableClass: 'table table-bordered plans-list',
  };

  return (
    <div>
      <Helmet>
        <title>{pageTitle}</title>
      </Helmet>
      <HeaderBreadcrumb {...breadcrumbProps} />
      <Row>
        <Col md={8}>
          <h3 className="mt-3 mb-3 page-title">{pageTitle}</h3>
        </Col>
      </Row>
      <Row>
        <Col>
          <ListView {...listViewProps} />
        </Col>
      </Row>
    </div>
  );
};

/** Declare default props for IRSPlansList */
const defaultProps: PlanListProps = {
  fetchPlans: fetchIRSPlans,
  plans: [],
  service: supersetFetch,
};

IRSPlansList.defaultProps = defaultProps;

export { IRSPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps {
  plans: IRSPlan[];
}

/** map state to props */
const mapStateToProps = (state: Partial<Store>): DispatchedStateProps => {
  const planDefinitionsArray = getIRSPlansArray(state);

  return {
    plans: planDefinitionsArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: fetchIRSPlans };

/** Connected ActiveFI component */
const ConnectedIRSPlansList = connect(
  mapStateToProps,
  mapDispatchToProps
)(IRSPlansList);

export default ConnectedIRSPlansList;
