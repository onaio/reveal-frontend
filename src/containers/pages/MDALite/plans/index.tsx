import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import {
  HIDDEN_PLAN_STATUSES,
  SUPERSET_MDA_LITE_REPORTING_PLANS_SLICE,
} from '../../../../configs/env';
import { MDA_LITE_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_LITE_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPlansReducer, {
  genericFetchPlans,
  GenericPlan,
  makeGenericPlansArraySelector,
  reducerName as genericReducerName,
} from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(genericReducerName, GenericPlansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for MDA lite plans */
const makeMDALitePlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a preview list of MDA plans */
const MDALitePlansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDALitePlansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: MDA_LITE_PLANS,
  pageUrl: REPORT_MDA_LITE_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_MDA_LITE_REPORTING_PLANS_SLICE,
};

MDALitePlansList.defaultProps = defaultProps;

export { MDALitePlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const MDALitePlansArray = makeMDALitePlansArraySelector(state, {
    interventionTypes: [InterventionType.MDALite],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans: MDALitePlansArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected ActiveFI component */
const ConnectedMDALitePlansList = connect(mapStateToProps, mapDispatchToProps)(MDALitePlansList);

export default ConnectedMDALitePlansList;
