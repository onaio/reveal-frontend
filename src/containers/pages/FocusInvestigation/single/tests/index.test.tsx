import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { FI_SINGLE_URL } from '../../../../../constants';
import store from '../../../../../store';
import { fetchGoals } from '../../../../../store/ducks/goals';
import { fetchJurisdictions } from '../../../../../store/ducks/jurisdictions';
import { fetchPlans } from '../../../../../store/ducks/plans';
import * as fixtures from '../../../../../store/ducks/tests/fixtures';
import ConnectedSingleFI, { SingleFI } from '../../single';

jest.mock('../../../../../components/GisidaWrapper', () => {
  const GisidaWrapperMock = () => <div>I love oov</div>;
  return GisidaWrapperMock;
});

jest.mock('../../../../../configs/env');

const history = createBrowserHistory();

describe('containers/pages/SingleFI', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const mock: any = jest.fn();
    const props = {
      goalsArray: fixtures.plan1Goals,
      history,
      jurisdiction: fixtures.jurisdictions[0],
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.plan1.id },
        path: `${FI_SINGLE_URL}/:id`,
        url: `${FI_SINGLE_URL}/13`,
      },
      planById: fixtures.plan1,
      plansArray: fixtures.plans,
      plansIdArray: fixtures.plansIdArray,
    };
    shallow(
      <Router history={history}>
        <SingleFI {...props} />
      </Router>
    );
  });

  it('renders SingleFI correctly', () => {
    const mock: any = jest.fn();
    const supersetMock: any = jest.fn();
    supersetMock.mockImplementation(() => Promise.resolve(fixtures.plans));
    const props = {
      goalsArray: fixtures.plan1Goals,
      history,
      jurisdiction: fixtures.jurisdictions[0],
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.plan1.id },
        path: `${FI_SINGLE_URL}/:id`,
        url: `${FI_SINGLE_URL}/16`,
      },
      planById: fixtures.plan1,
      plansArray: fixtures.plans,
      plansIdArray: fixtures.plansIdArray,
      supersetService: supersetMock,
    };
    const wrapper = mount(
      <Router history={history}>
        <SingleFI {...props} />
      </Router>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('GisidaWrapperMock').props()).toMatchSnapshot();
    expect(wrapper.find('HeaderBreadcrumb').length).toEqual(1);
    wrapper.unmount();
  });

  it('renders SingleFI correctly for null plan jurisdction id', () => {
    const mock: any = jest.fn();
    const supersetMock: any = jest.fn();
    supersetMock.mockImplementation(() => Promise.resolve(fixtures.plan5));
    const props = {
      goalsArray: [],
      history,
      jurisdiction: fixtures.jurisdictions[0],
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.plan5.id },
        path: `${FI_SINGLE_URL}/:id`,
        url: `${FI_SINGLE_URL}/16`,
      },
      planById: fixtures.plan5,
      plansArray: [fixtures.plan5],
      plansIdArray: [fixtures.plan5.id],
      supersetService: supersetMock,
    };
    const wrapper = mount(
      <Router history={history}>
        <SingleFI {...props} />
      </Router>
    );
    expect(wrapper.find('GisidaWrapperMock').length).toEqual(0);
    expect(wrapper.find('HeaderBreadcrumb').length).toEqual(1);
    wrapper.unmount();
  });

  it('It works with the Redux store', () => {
    const mock: any = jest.fn();
    store.dispatch(fetchPlans(fixtures.plans));
    store.dispatch(fetchGoals(fixtures.goals));
    store.dispatch(fetchJurisdictions(fixtures.jurisdictions));
    const props = {
      history,
      jurisdiction: fixtures.jurisdictions[0],
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.plan1.id },
        path: `${FI_SINGLE_URL}/:id`,
        url: `${FI_SINGLE_URL}/16`,
      },
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedSingleFI {...props} />
        </Router>
      </Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('GisidaWrapperMock').props()).toMatchSnapshot();
    wrapper.unmount();
  });
});
