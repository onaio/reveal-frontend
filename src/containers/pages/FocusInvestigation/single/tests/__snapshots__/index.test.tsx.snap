// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/SingleFI It works with the Redux store 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(SingleFI)
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={[MockFunction]}
      match={
        Object {
          "isExact": true,
          "params": Object {
            "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
          },
          "path": "/focus-investigation/view/:id",
          "url": "/focus-investigation/view/16",
        }
      }
    >
      <SingleFI
        fetchGoalsActionCreator={[Function]}
        fetchPlansActionCreator={[Function]}
        goalsArray={Array []}
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={[MockFunction]}
        match={
          Object {
            "isExact": true,
            "params": Object {
              "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
            },
            "path": "/focus-investigation/view/:id",
            "url": "/focus-investigation/view/16",
          }
        }
        planById={
          Object {
            "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
            "jurisdiction_depth": 2,
            "jurisdiction_id": "3377",
            "jurisdiction_name": "NVI_439",
            "jurisdiction_name_path": Array [
              "Chadiza",
              "Naviluli",
            ],
            "jurisdiction_parent_id": "2944",
            "jurisdiction_path": Array [
              "2939",
              "2944",
            ],
            "plan_fi_reason": "Routine",
            "plan_fi_status": "A1",
            "plan_id": "10f9e9fa-ce34-4b27-a961-72fab5206ab6",
            "plan_status": "active",
            "plan_title": "A1-Tha Luang Village 1 Focus 01",
          }
        }
        plansArray={
          Array [
            Object {
              "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
              "jurisdiction_depth": 2,
              "jurisdiction_id": "3377",
              "jurisdiction_name": "NVI_439",
              "jurisdiction_name_path": Array [
                "Chadiza",
                "Naviluli",
              ],
              "jurisdiction_parent_id": "2944",
              "jurisdiction_path": Array [
                "2939",
                "2944",
              ],
              "plan_fi_reason": "Routine",
              "plan_fi_status": "A1",
              "plan_id": "10f9e9fa-ce34-4b27-a961-72fab5206ab6",
              "plan_status": "active",
              "plan_title": "A1-Tha Luang Village 1 Focus 01",
            },
            Object {
              "id": "plan-id-2",
              "jurisdiction_depth": 2,
              "jurisdiction_id": "3378",
              "jurisdiction_name": "TLv1_02",
              "jurisdiction_name_path": Array [
                "Canton Tha Luang",
                "Tha Luang Village",
              ],
              "jurisdiction_parent_id": "2977",
              "jurisdiction_path": Array [
                "2989",
                "2977",
              ],
              "plan_fi_reason": "Case-triggered",
              "plan_fi_status": "A1",
              "plan_id": "plan-id-2",
              "plan_status": "active",
              "plan_title": "A1-Tha Luang Village 1 Focus 01",
            },
          ]
        }
        plansIdArray={
          Array [
            "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
            "plan-id-2",
          ]
        }
      >
        <div>
          <h2
            className="page-title mt-4 mb-5"
          >
            Focus Investigation
             
            in
             
            NVI_439
          </h2>
          <Row
            tag="div"
          >
            <div
              className="row"
            >
              <Col
                className="col-6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-6 col"
                >
                  <h4
                    className="mb-4"
                  >
                    Focus Area Information
                  </h4>
                  <div
                    style={
                      Object {
                        "background": "lightgrey",
                        "height": "200px",
                      }
                    }
                  />
                  <dl
                    className="row mt-3"
                  >
                    <dt
                      className="col-5"
                    >
                      District
                    </dt>
                    <dd
                      className="col-7"
                    />
                    <dt
                      className="col-5"
                    >
                      Canton
                    </dt>
                    <dd
                      className="col-7"
                    >
                      Chadiza
                    </dd>
                    <dt
                      className="col-5"
                    >
                      FI Reason
                    </dt>
                    <dd
                      className="col-7"
                    >
                      Routine
                    </dd>
                    <dt
                      className="col-5"
                    >
                      FI Status
                    </dt>
                    <dd
                      className="col-7"
                    >
                      A1
                    </dd>
                  </dl>
                  <hr />
                </div>
              </Col>
              <Col
                className="col-6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-6 col"
                >
                  <div
                    className="fi-active"
                  >
                    <h5
                      className="mb-4 mt-1"
                    >
                      Active Investigation
                    </h5>
                    <dl
                      className="row mt-3"
                    >
                      <dt
                        className="col-5"
                      >
                        Complete
                      </dt>
                      <dd
                        className="col-7"
                      >
                        No
                      </dd>
                    </dl>
                    <hr />
                    <h5
                      className="mb-4 mt-4"
                    >
                      Response
                    </h5>
                  </div>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </SingleFI>
    </Connect(SingleFI)>
  </Router>
</Provider>
`;

exports[`containers/pages/SingleFI renders SingleFI correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <SingleFI
    fetchGoalsActionCreator={[Function]}
    fetchPlansActionCreator={[Function]}
    goalsArray={
      Array [
        Object {
          "action_code": "Case Confirmation",
          "completed_task_count": 0,
          "goal_comparator": "=",
          "goal_id": "Case_Confirmation",
          "goal_unit": "each",
          "goal_value": 1,
          "id": "5a27ec10-7a5f-563c-ba11-4de150b336af",
          "jurisdiction_id": "450fc15b-5bd2-468a-927a-49cb10d3bcac",
          "measure": "Case confirmation complete",
          "plan_id": "10f9e9fa-ce34-4b27-a961-72fab5206ab6",
          "task_count": 0,
        },
        Object {
          "action_code": "Larval Dipping",
          "completed_task_count": 0,
          "goal_comparator": ">=",
          "goal_id": "Larval_Dipping_Min_3_Sites",
          "goal_unit": "each",
          "goal_value": 3,
          "id": "f8d4e0a9-5867-5c78-9e26-de45d72556c4",
          "jurisdiction_id": "450fc15b-5bd2-468a-927a-49cb10d3bcac",
          "measure": "Number of larval dipping activities",
          "plan_id": "10f9e9fa-ce34-4b27-a961-72fab5206ab6",
          "task_count": 2,
        },
      ]
    }
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    location={[MockFunction]}
    match={
      Object {
        "isExact": true,
        "params": Object {
          "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
        },
        "path": "/focus-investigation/view/:id",
        "url": "/focus-investigation/view/16",
      }
    }
    planById={
      Object {
        "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
        "jurisdiction_depth": 2,
        "jurisdiction_id": "3377",
        "jurisdiction_name": "NVI_439",
        "jurisdiction_name_path": Array [
          "Chadiza",
          "Naviluli",
        ],
        "jurisdiction_parent_id": "2944",
        "jurisdiction_path": Array [
          "2939",
          "2944",
        ],
        "plan_fi_reason": "Routine",
        "plan_fi_status": "A1",
        "plan_id": "10f9e9fa-ce34-4b27-a961-72fab5206ab6",
        "plan_status": "active",
        "plan_title": "A1-Tha Luang Village 1 Focus 01",
      }
    }
    plansArray={
      Array [
        Object {
          "id": "ed2b4b7c-3388-53d9-b9f6-6a19d1ffde1f",
          "jurisdiction_depth": 2,
          "jurisdiction_id": "3377",
          "jurisdiction_name": "NVI_439",
          "jurisdiction_name_path": Array [
            "Chadiza",
            "Naviluli",
          ],
          "jurisdiction_parent_id": "2944",
          "jurisdiction_path": Array [
            "2939",
            "2944",
          ],
          "plan_fi_reason": "Routine",
          "plan_fi_status": "A1",
          "plan_id": "10f9e9fa-ce34-4b27-a961-72fab5206ab6",
          "plan_status": "active",
          "plan_title": "A1-Tha Luang Village 1 Focus 01",
        },
        Object {
          "id": "plan-id-2",
          "jurisdiction_depth": 2,
          "jurisdiction_id": "3378",
          "jurisdiction_name": "TLv1_02",
          "jurisdiction_name_path": Array [
            "Canton Tha Luang",
            "Tha Luang Village",
          ],
          "jurisdiction_parent_id": "2977",
          "jurisdiction_path": Array [
            "2989",
            "2977",
          ],
          "plan_fi_reason": "Case-triggered",
          "plan_fi_status": "A1",
          "plan_id": "plan-id-2",
          "plan_status": "active",
          "plan_title": "A1-Tha Luang Village 1 Focus 01",
        },
      ]
    }
    plansIdArray={
      Array [
        "10f9e9fa-ce34-4b27-a961-72fab5206ab6",
      ]
    }
  >
    <div>
      <h2
        className="page-title mt-4 mb-5"
      >
        Focus Investigation
         
        in
         
        NVI_439
      </h2>
      <Row
        tag="div"
      >
        <div
          className="row"
        >
          <Col
            className="col-6"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col-6 col"
            >
              <h4
                className="mb-4"
              >
                Focus Area Information
              </h4>
              <div
                style={
                  Object {
                    "background": "lightgrey",
                    "height": "200px",
                  }
                }
              />
              <dl
                className="row mt-3"
              >
                <dt
                  className="col-5"
                >
                  District
                </dt>
                <dd
                  className="col-7"
                />
                <dt
                  className="col-5"
                >
                  Canton
                </dt>
                <dd
                  className="col-7"
                >
                  Chadiza
                </dd>
                <dt
                  className="col-5"
                >
                  FI Reason
                </dt>
                <dd
                  className="col-7"
                >
                  Routine
                </dd>
                <dt
                  className="col-5"
                >
                  FI Status
                </dt>
                <dd
                  className="col-7"
                >
                  A1
                </dd>
              </dl>
              <hr />
            </div>
          </Col>
          <Col
            className="col-6"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col-6 col"
            >
              <div
                className="fi-active"
              >
                <h5
                  className="mb-4 mt-1"
                >
                  Active Investigation
                </h5>
                <dl
                  className="row mt-3"
                >
                  <dt
                    className="col-5"
                  >
                    Complete
                  </dt>
                  <dd
                    className="col-7"
                  >
                    No
                  </dd>
                </dl>
                <hr />
                <h5
                  className="mb-4 mt-4"
                >
                  Response
                </h5>
                <div
                  className="responseItem"
                  key="Case_Confirmation"
                >
                  <h6>
                    Case Confirmation
                  </h6>
                  <div
                    className="targetItem"
                  >
                    <p>
                      Measure
                      : 
                      Case confirmation complete
                    </p>
                    <p>
                      Target
                      : 
                      0
                       
                      of
                       
                      1
                    </p>
                    <ProgressBar
                      decimalPoints={0}
                      height="10px"
                      max={1}
                      min={0}
                      value={0}
                    >
                      <div
                        className="progress"
                        style={
                          Object {
                            "height": "10px",
                            "marginBottom": "15px",
                          }
                        }
                      >
                        <div
                          aria-valuemax={1}
                          aria-valuemin={0}
                          aria-valuenow={0}
                          className="progress-bar"
                          role="progressbar"
                          style={
                            Object {
                              "backgroundColor": "#ff3",
                              "width": "0%",
                            }
                          }
                        />
                      </div>
                    </ProgressBar>
                  </div>
                </div>
                <div
                  className="responseItem"
                  key="Larval_Dipping_Min_3_Sites"
                >
                  <h6>
                    Larval Dipping
                  </h6>
                  <div
                    className="targetItem"
                  >
                    <p>
                      Measure
                      : 
                      Number of larval dipping activities
                    </p>
                    <p>
                      Target
                      : 
                      2
                       
                      of
                       
                      3
                    </p>
                    <ProgressBar
                      decimalPoints={0}
                      height="10px"
                      max={3}
                      min={0}
                      value={2}
                    >
                      <div
                        className="progress"
                        style={
                          Object {
                            "height": "10px",
                            "marginBottom": "15px",
                          }
                        }
                      >
                        <div
                          aria-valuemax={3}
                          aria-valuemin={0}
                          aria-valuenow={66.66666666666666}
                          className="progress-bar"
                          role="progressbar"
                          style={
                            Object {
                              "backgroundColor": "#ff5c33",
                              "width": "67%",
                            }
                          }
                        />
                      </div>
                    </ProgressBar>
                  </div>
                </div>
              </div>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </SingleFI>
</Router>
`;
